packages/schematics/angular/app-shell/index.ts

```diff
@@ -66,6 +66,13 @@ function getServerModulePath(host: Tree, app: AppConfig): string | null {
 
 function addUniversalApp(options: AppShellOptions): Rule {
   return (host: Tree, context: SchematicContext) => {
+    const config = getConfig(host);
+    const appConfig = getAppFromConfig(config, options.universalApp);
+
+    if (appConfig && appConfig.platform === 'server') {
```

Comment:
If `platform !== 'server'` but it exists, that should throw an exception.

Reply:
Additional check is added to throw for the invalid option value.